web-api/tests/personalized_semantic_search.rs

```diff
@@ -79,18 +79,28 @@ struct PersonalizedDocumentData {
     properties: serde_json::Value,
 }
 
-#[derive(Deserialize)]
+#[derive(Debug, Deserialize)]
 struct SemanticSearchResponse {
     documents: Vec<PersonalizedDocumentData>,
 }
 
-impl SemanticSearchResponse {
-    fn ids(&self) -> Vec<&str> {
-        self.documents
-            .iter()
-            .map(|document| document.id.as_str())
-            .collect_vec()
-    }
+macro_rules! assert_order {
+    ($documents: expr, $ids: expr, $($arg: tt)*) => {
+        assert_eq!(
+            $documents
+                .iter()
+                .map(|document| document.id.as_str())
+                .collect_vec(),
+            $ids,
+            $($arg)*
+        );
+        for documents in $documents.windows(2) {
+            let [d1, d2] = documents else { unreachable!() };
+            assert!(1.0 >= d1.score, $($arg)*);
+            assert!(d1.score > d2.score, $($arg)*);
+            assert!(d2.score >= 0.0, $($arg)*);
+        }
+    };
```

Comment:
:art: You could move it into the integration test utils, if it's used in multiple integration test builds.

Reply:
i think there are some code duplications already. we can clean it up in a following pr. some parts may not be moved to the integration test utils crate, as this can cause cyclic dependencies depending on the used types. but we can always have a common utils module in the web api integration test directory which can be imported in each integration test binary.

Reply:
> can cause cyclic dependencies depending on the used typ

This is why I initially places the tests in the `integration-tests` crate (also this is why it's named `integration-tests` instead of e.g. `integration-test-utils`).